### ISSUE TITLE
Want to install parallel coordinates in Orange3

### ISSUE THREAD

**Issue by SamGG** [biolab/orange3#2310]
Hi,
I would like to use the parallel coordinates widget. This was present in V2.7 (if I remember well), but I didn't find it in V3.4. I found the code in GitHub. Could you help me in adding this visualization widget in V3.4 under Windows?
Best.
https://github.com/biolab/orange3/blob/7dca99338f1eaca7f8d6f4925dd79944a8aa426d/Orange/widgets/visualize/owparallelcoordinates.py
I already asked on stackexchange as advised, but no answer yet.
https://datascience.stackexchange.com/questions/18816/
I did successfully install network tools using tar.gz, pip install and MVS compiler.
Best.

##### Orange version
Orange is 3.4.2, git revision f3076cb, Python 3.4.4 (v3.4.4:737efcadf5a6, Dec 20 2015, 19:28:18) [MSC v.1600 32 bit (Intel)] on win32
running on Win7 Pro 64 bits

##### Expected behavior
I want to view data using parallel coordinates

##### Actual behavior
Not available, although the widget is present on GitHub

##### Steps to reproduce the behavior
NA

##### Additional info (worksheets, data, screenshots, ...)
NA

**Comment by kernc** [comment]
You can checkout https://github.com/biolab/orange3/pull/2327 and see if it works for your use case. Thanks!

**Comment by SamGG** [comment]
Thanks.

**Comment by lukaseller** [comment]
I am also wondering how to get "Parallel Coordinates" up and running. 
Already installed Prototypes 0.14.2 for Orange 3.27 on MacOs, but i cannot find it. 
Was it moved to another package? 
Thanks in advance

**Comment by rawiz-coder** [comment]
I was hoping to use this too but also can't see it in Mac OS Orange v3.39.0 and Prototypes v0.21.2, is it not there anymore or somewhere else or just obsolete (any plans to bring it back)? This kind of visualisation is really great for say eyeballing what kind of factory conditions lead to particular things. Maybe some other visualisation is suggested instead though?

**Comment by wvdvegte** [comment]
Parallel coordinates plot is a very common visualization in machine learning. I don't understand why it's (currently) not available in Orange.

**Comment by rawiz-coder** [comment]
Interestingly it does appear to be in KNIME, although ORANGE is supposed to be easier to use of course.

**Comment by wvdvegte** [comment]
Indeed, KNIME has a node for Parallel Coordinates Plot. KNIME offers a lot of data operations that are not available in Orange And yes, at least in my experience, Orange is easier to use. For instance, in Orange you can use one Edit Domain widget to convert a lot of variables into other types of variables, e.g., a date into a number, a categorical variable into text, etc., while in KNIME every type of conversion has its own node (equivalent of widget) - e.g., date to number node, category to text node, etc.
KNIME also has support for looping a workflow over a variable - which by itself also introduces a lot of extra complexity. In Orange, this is not possible.
I recently played around with KNIME but concluded that it cannot replace Orange in my field - education in data analytics and ML to non-CS students. On the other hand, there seems to be a large lobby (mainly CS people) advocating that these topics should be taught and practiced using Python code snippets in a notebook environment.

**Comment by janezd** [comment]
I don't think Parallel coordinates are coming back. Orange 2 had them, and so did Orange 3 as a prototype, but it was based on plotly and it was unreliable and unmaintained.

The problem with Parallel coordinates, in my opinion, is that it requires or implies a certain order of variables. The visual effect of the visualization depends on it, yet there is most often no "natural ordering". The least one has to do is add an algorithm for ordering; we had one in Orange 2, but we didn't find it very useful.

That said, if somebody implements a proper widget (which would follow all the unwritten standards of Orange GUI and coding :( ), I don't think there'd be any opposition to adding it. On the other hand, I don't think any of the common contributors is fond of this visualization and won't invest time into implementing it.

**Comment by rawiz-coder** [comment]
It’s easy for factory data, you just order the variables as you flow through the process. This is a write up of it in a commercial time series factory data tool (others are available), incredibly useful for seeing how different modes and operating envelopes lead to different performance:

https://seeq12.github.io/seeq-parallelcoordinates/user_guide.html

Sometimes it’s good to have the output quality variables on the far right of course.
